docs/manual/java/guide/broker/MessageBrokerApi.md

```diff
@@ -34,6 +34,12 @@ Here's an example of publishing a single, non sharded event stream:
 
 Note that the read-side event stream you passed to the topic producer is "activated" as soon as the service is started. That means all events persisted by your services will eventually be published to the connected topic. Also, you will generally want to map your domain events into some other type, so that other service won't be tightly coupled to another service's domain model events. In other words, domain model events are an implementation detail of the service, and should not be leaked.
 
+### Filtering events
+
+You may not want all events persisted by your services to be published. If that is the case then you can filter the event stream:
+
+@[filter-events](code/docs/javadsl/mb/FilteredServiceImpl.java)
+
```

Comment:
This is unclear wrt the implication of filtering: when an event from `eventStream` is filtered (not produced to Kafka)  the `offset` is not forwarded. This is usually fine then the code in the `TopicProducer` doesn't generate side effects, the % of filtered messages is low or is not time/CPU-consuming. Not advancing the `offset` means messages could be consumed form the stream twice.

In general, `TopicProducer` may consume events twice since `offset` is not committed on every event but filtering  increases the amount of event consumption that will happen.